runtime/lua/vim/lsp/buf.lua

```diff
@@ -827,11 +827,14 @@ local function on_code_action_results(results, opts)
     return
   end
 
-  ---@param item {action: lsp.Command|lsp.CodeAction}
+  ---@param item {action: lsp.Command|lsp.CodeAction, ctx : lsp.HandlerContext}
   local function format_item(item)
-    local title = item.action.title:gsub('\r\n', '\\r\\n')
-    return title:gsub('\n', '\\n')
+    local source = vim.lsp.get_client_by_id(item.ctx.client_id).name or 'unknown'
+    return ([[%s [%s] ]]):format(item.action.title, source)
```

Comment:
Not sure if we really need this fallback, and if 'unknown' is a good one to display. Do we have any particular path where `vim.lsp.Client.name` can be nil? (it's a non-nil field).

Also please prefer `''` instead of `[[ ]]` here because it reads (slightly) better. Q: why do we need a trailing whitespace here?

Reply:
Trailing whitespace was due to previous impl. `[[%s [%s]]]` is wanted, but trying to do that gave me errors. Using `''` seems to fix this though.